internal/release/promote/promotion.go

```diff
@@ -214,3 +209,28 @@ func getTargetEnvironments(environments []*v1alpha1.Environment, sourceEnvironme
 	}
 	return envs, nil
 }
+
+func (p *Promotion) selectCreatingPR(opts *Opts) (bool, error) {
```

Comment:
I am not sure I like the division of this function.
What I don't like:
- It does the prompting, and modifies the opts pointer. Modifying the opts from this func might be unexpected to the caller.
- It introduces a cancel flag which is unneeded if you don't break this function out.